bofire/strategies/doe/objective.py

```diff
@@ -474,6 +488,205 @@ def _convert_input_to_tensor(
         return torch.tensor(X.values, requires_grad=requires_grad, **tkwargs)
 
 
+class IOptimality(Objective):
+    """A class implementing the evaluation of I-criterion and its jacobian w.r.t. the inputs."""
+
+    def __init__(
+        self,
+        domain: Domain,
+        model: Formula,
+        n_experiments: int,
+        delta: float = 1e-6,
+        transform_range: Optional[Bounds] = None,
+        n_space: Optional[int] = None,
+        ipopt_options: Optional[dict] = None,
+    ) -> None:
+        """
+        Args:
+            domain (Domain): A domain defining the DoE domain together with model_type.
+            model_type (str or Formula): A formula containing all model terms.
+            n_experiments (int): Number of experiments
+            delta (float): A regularization parameter for the information matrix. Default value is 1e-3.
+            transform_range (Bounds, optional): range to which the input variables are transformed before applying the objective function. Default is None.
+            n_space (int, optional): Number of space filling points. If none is provided,
+                n_space = n_experiments is assumed. Only relevant if SpaceFilling is used
+                to fill the feasible space, i.e. in presence of equality constraints.
+                Otherwise a uniform grid is generated. If None is provided, n_space = 10 * n_experiments is assumed.
+                Default is None.
+            ipopt_options (dict, optional): Options for the Ipopt solver to generate space filling point.
+                If None is provided, the default options (maxiter = 500) are used.
+        """
+
+        if transform_range is not None:
+            raise ValueError(
+                "IOptimality does not support transformations of the input variables."
+            )
+
+        super().__init__(
+            domain=domain,
+            model=model,
+            n_experiments=n_experiments,
+            delta=delta,
+            transform_range=transform_range,
+        )
+
+        # uniformly fill the design space
+        if np.any([isinstance(obj, EqualityConstraint) for obj in domain.constraints]):
+            warnings.warn(
+                "Equality constraints were detected. No equidistant grid of points can be generated. The design space will be filled via SpaceFilling.",
+                UserWarning,
+            )
+            if n_space is None:
+                n_space = n_experiments * 10
+
+            print(
```

Comment:
We should in a seperate PR introduce some kind of general logger and use it everywhere.